File: probs/1.cpp
#include <vector>
#include <map>
#include <iostream>
using namespace std;

vector<int> twoSum(vector<int> &nums, int target)
{
    map<int, int> mp;

    for (int i = 0; i < nums.size(); i++)
    {
        mp[nums[i]] = i;
    }

    for (int i = 0; i < nums.size(); i++)
    {
        int t = target - nums[i];

        if (mp.find(t) != mp.end() && mp[t] != i)
        {
            return {i, mp[t]};
        }
    }

    return {};
}

int main()
{
    vector<int> vec = {3, 2, 4};

    vector<int> result = twoSum(vec, 6);

    cout << result[0] << " " << result[1] << "\n";

    return 0;
}

File: go/1.go
package main

func twoSum(nums []int, target int) []int {
	mp := map[int]int{}

	for idx, val := range nums {
		mp[val] = idx
	}

	for idx, num := range nums {
		pair, ok := mp[target-num]

		if ok && pair != idx {
			return []int{idx, pair}
		}
	}

	return []int{}
}
